submitit/helpers.py

```diff
@@ -216,3 +217,67 @@ def __enter__(self) -> None:
 
     def __exit__(self, *args):
         os.chdir(self.original_dir)
+
+
+def monitor_jobs(jobs: tp.Sequence[core.Job[core.R]], sleep_time_s: float = 30, test_mode: bool = False):
+    """Continuously monitors given jobs until they are all done or failed.
+
+    Parameters
+    ----------
+    jobs: List[Jobs]
+        A list of jobs to monitor
+    sleep_time_s: int
+        The time (in seconds) between two refreshes of the monitoring.
+        Can't be inferior to 30s.
+    test_mode: bool
+        If in test mode, we do not check the length of sleep_time_s
+
+    Yields
+    ------
+    done: set
+        The set of jobs that has succeeded
+    failed_jobs: set
+        The set of jobs that has failed
+    """
+
+    if not test_mode:
+        assert sleep_time_s >= 30, "You can't refresh too often (>= 30s) to avoid overloading squeue"
+
+    monitoring_start_time = time.time()
+    n_jobs = len(jobs)
+    done, failed_jobs = set(), set()
+    while True:
+        state_count = {"PENDING": 0, "RUNNING": 0}
+        for i, job in enumerate(jobs):
+            if i in done or i in failed_jobs:
+                continue
+            job_state = job.state.upper()
+            if job_state in state_count:
+                state_count[job_state] += 1
+            elif job_state == "FAILED":
+                failed_jobs.add(i)
+            elif job.done():
+                done.add(i)
```

Comment:
Simpler proposition:
```suggestion 
    state_jobs = collections.defaultdict(set)
    while True:
        for state in ["PENDING", "RUNNING"]"
            state_jobs[state] = set()
        for i, job in enumerate(jobs):
            state_jobs[job.state.upper()].add(i)
```
You get more information that youo need, but nevermind it's just simpler